src/to_device.jl

```diff
@@ -0,0 +1,60 @@
+import Adapt
+import ClimaComms
+
+"""
+    out = to_device(device, x::Union{
```

Comment:
```suggestion
    out = to_device(device::ClimaComms.AbstractDevice, x::Union{
```

Reply:
I don't think I've ever seen docstrings with a named return value

Reply:
I've occasionally seen them. I feel that it can sometimes help explain them in the remaining docstring.

Reply:
Yes, I agree.